### PR TITLE
sniffer: extend script to support sockets

### PR DESCRIPTION
**Pull request by cgundogan**
This PR extends the python sniffer script to support sockets.
You can test by flashing the sniffer application to a node on the iot-lab testbed and bind it to localhost with:
```
ssh -L 20000:<node-id>:20000 <user>@<site>.iot-lab.info
```
Now you can call:
```bash
python2 sniffer.py socket localhost 20000 26 | wireshark -k -i -
```

### CODE REVIEW

**Comment by OlegHahm** [comment]
nice

**Comment by authmillenon** [comment]
How do I get the sniffer app to open a socket?

**Comment by authmillenon** [comment]
(on the testbed e.g.)

**Comment by authmillenon** [comment]
Ah… it's in the readme... sorry for the noise.

File: dist/tools/sniffer/sniffer.py [hunk 1]
```diff
@@ -104,28 +107,46 @@ def generate_pcap(port, out):
...
+
+        try:
+            sock = socket.socket()
+            sock.connect((host, port))
+            conn = sock.makefile("r+b", bufsize=0)
+        except IOError:
+            print("error connecting to %s:%s" % (host, port), file=sys.stderr)
+            sys.exit(2)
+    else:
+        print("error: unsupported connection type. Use \"serial\" or \"socket\"")
```
Comment (authmillenon) [inline dist/tools/sniffer/sniffer.py]: Might be a good idea to put all this into its own function.

**Comment by authmillenon** [comment]
Apart from comment: ACK. Tested it and it worked :-)

**Comment by cgundogan** [comment]
I used `gnrc_example` instead of `gnrc_networking` in the README. I changed this and amended immediately

**Comment by authmillenon** [comment]
What about my comment?

**Comment by cgundogan** [comment]
@authmillenon addressed your comment

**Comment by authmillenon** [comment]
re-ACK, please squash.

**Comment by cgundogan** [comment]
squashed

**Comment by cgundogan** [comment]
travis agrees => GO

**Pull request merged**